### PR TITLE
feat(ONL-4295): Adding back button to EcPanel

### PR DESCRIPTION
**Pull request by jafcalvente**
- This story is for adding to the EcPanel the possibility of displaying (is optional) a new button in the top-left corner, with a left arrow icon.

- This button just will emit an event to point out that user wants to go back to a previous panel.

- To display it in Storybook, go to Panel basic component, and mark "With the back button visible" in the Knobs tab. 

image

### CODE REVIEW

File: src/components/ec-panel/ec-panel.spec.js [hunk 1]
```diff
@@ -38,15 +38,38 @@ describe('EcPanel', () => {
       expect(wrapper.find('.ec-panel').exists()).toBe(false);
       expect(wrapper.element).toMatchSnapshot();
     });
+
+    it(':goBackEnabled - should render the back button within the panel when is true', () => {
+      const wrapper = mountPanel({ show: true, goBackEnabled: true });
```
Comment (matallo) [inline src/components/ec-panel/ec-panel.spec.js]: these props shoudl be named `isSomething`, ask @jaimeperezpedra for more context

Comment (jafcalvente) [reply src/components/ec-panel/ec-panel.spec.js]: Amended.

File: src/components/ec-panel/ec-panel.vue [hunk 2]
```diff
@@ -36,24 +49,37 @@ export default {
...
     prop: 'show',
-    event: 'close',
+    event: 'show-panel',
   },
   props: {
     show: {
       type: Boolean,
       default: false,
     },
+    isBackEnabled: {
```
Comment (mcibique) [inline src/components/ec-panel/ec-panel.vue]: I wouldn't make this as a prop. I would convert it to a computed prop and check if $listeners['back'] exists inside. Then everything you have to do to enable back functionality is to `<ec-panel @back="doSomething()">`

File: src/components/ec-panel/ec-panel.vue [hunk 3]
```diff
@@ -5,14 +5,27 @@
...
+          <ec-icon
+            class="ec-panel__header-icon"
+            name="simple-chevron-left"
+            :size="24"
+          />
+        </a>
         <a
           aria-label="Close panel"
-          class="ec-panel__close"
+          class="ec-panel__header-action ec-panel__header-action--close"
```
Comment (mcibique) [inline src/components/ec-panel/ec-panel.vue]: OT: While you are here, could you please add data-test on everything clickable in this component? We have lots of references to these classes in integration tests and unit tests, we should replace them with data-test.

Comment (jafcalvente) [reply src/components/ec-panel/ec-panel.vue]: Well spotted. Done.

File: src/components/ec-panel/ec-panel.vue (hunk 2 above)
Comment (jafcalvente) [reply src/components/ec-panel/ec-panel.vue]: Smarter approach. Done.

**Pull request merged**